### PR TITLE
Vizotenko/tdi 41626

### PR DESCRIPTION
**Pull request by vizotenko-talend**
[TDI-41626
[tCreateTemporaryFile] Safely create a temporary file](https://jira.talendforge.org/browse/TDI-41626)

### CODE REVIEW

File: main/plugins/org.talend.repository/plugin.xml [hunk 1]
```diff
@@ -3155,6 +3155,15 @@
               name="ResetAdditionalJDBCParamsForCreateTable"
               version="7.2.1">
         </projecttask>
+       <projecttask
+               beforeLogon="false"
+               breaks="7.2.0"
+               class="org.talend.repository.model.migration.SafelyCreateTemporaryFileMigrationTask"
+               description="Add conf"
```
Comment (ypiel-talend) [inline main/plugins/org.talend.repository/plugin.xml]: The description is too vague. It should be something : "Force old behaviour for existing jobs, for create temporary files" or something like this.

File: main/plugins/org.talend.designer.components.localprovider/components/tCreateTemporaryFile/tCreateTemporaryFile_java.xml [hunk 2]
```diff
@@ -56,23 +56,44 @@
...
+
     <PARAMETER
       NAME="INFO_RANGE"
       FIELD="LABEL"
       REQUIRED="false"
       NUM_ROW="40"
+      SHOW_IF="USE_PREFIX == 'false'"
      >
-      <DEFAULT>the temporary file name should include template "XXXX".</DEFAULT>
+      <DEFAULT>the temporary file name should include template "XXXX". Template is deprecated. Use Prefix instead.</DEFAULT>
```
Comment (ypiel-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/tCreateTemporaryFile/tCreateTemporaryFile_java.xml]: Set the first letter to uppercase (not your code but should have been like this before).

Comment (vizotenko-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tCreateTemporaryFile/tCreateTemporaryFile_java.xml]: done

File: main/plugins/org.talend.repository/plugin.xml (hunk 1 above)
Comment (vizotenko-talend) [reply main/plugins/org.talend.repository/plugin.xml]: done

**Pull request merged**